apps/src/lib/util/javascriptMode.js

```diff
@@ -65,11 +65,15 @@ export function apiValidateType(
         break;
       case 'pinid':
         var reservedPins = ['A2', 'A3', 'A7', 1, 9, 10];
-        properType =
-          (typeof varValue === 'string' || typeof varValue === 'number') &&
-          !reservedPins.includes(varValue);
+        var validPins = [0, 2, 3, 6, 12, 'A0', 'A1', 'A4', 'A5', 'A6'];
+        properType = validPins.includes(varValue);
         if (reservedPins.includes(varValue)) {
```

Comment:
I feel like we'd really only care if they used reserved pins if we already knew that the pin was definitely valid. Swapping this with the current `else if` clause would allow us to prioritize finding cases that should warrant an actual error.

Reply:
This makes a lot of sense! Swapped the order of conditions. Thanks!